scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -114,20 +114,29 @@ public <K> BaseAsyncLookupDoFn(int maxPendingRequests, CacheSupplier<A, B, K> ca
    *     same input
    * @param cacheSupplier supplier for lookup cache.
    */
-  public <K> BaseAsyncLookupDoFn(
-      int maxPendingRequests, boolean deduplicate, CacheSupplier<A, B, K> cacheSupplier) {
-    this.instanceId = UUID.randomUUID();
+  public BaseAsyncLookupDoFn(
+      int maxPendingRequests, boolean deduplicate, CacheSupplier<A, B> cacheSupplier) {
     this.cacheSupplier = cacheSupplier;
     this.deduplicate = deduplicate;
     this.semaphore = new Semaphore(maxPendingRequests);
   }
 
-  protected abstract C newClient();
+  @Override
+  public ResourceType getResourceType() {
+    return ResourceType.PER_CLONE;
+  }
```

Comment:
Maybe we should leave this abstract.
I was conservative here. Previous code looked to be `PER_INSTANCE` but this won't work if client is not thread safe

Reply:
Hmm.. maybe we should not override `getResourceType` here and let implementations be explicit about their level of resource-sharing dependent on client type, wdyt? I'm thinking about common use cases like adding rate-limiting, where going from per-instance-->per-clone could be dangerous to do opaquely to the user.